internal/manifests/mutate.go

```diff
@@ -194,7 +194,6 @@ func mutateConfigMap(existing, desired *corev1.ConfigMap) {
 }
 
 func mutateServiceAccount(existing, desired *corev1.ServiceAccount) {
-	existing.Annotations = desired.Annotations
```

Comment:
doesn't this mean that if a user were to add an annotation that is expected to propagate to their serviceaccount, we would no longer do this? Also, if we remove an annotation it will no longer be removed? should we do some type of merge here?

Reply:
> doesn't this mean that if a user were to add an annotation that is expected to propagate to their serviceaccount, we would no longer do this? 

This PR actually makes sure that the user-added annotations on SA are not reverted by the operator. The mutate.go by default keeps user-defined annotations but for some kinds those annotations are removed (e.g. SA) and this PR relaxes that for SA.

Reply:
yes, but what about annotations added to the desired SA? i.e. the current logic allows us to update the existing SA with the desired new annotations

Reply:
The logic in this PR makes sure that the desired annotations (set by the operator) are enforced and the user-defined annotations are not removed.

Reply:
can you add a test for the serviceaccount annotation updating to verify that in [here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/controllers/reconcile_test.go#L77)?

Reply:
I have added a test